### PR TITLE
Add development docs

### PR DESCRIPTION
**Pull request by novanai**
Add developer documentation, explaining the different libraries we use and how to understand their individual documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,53 +1,8 @@
 # Blockbot
 
-Blockbot is a Discord bot, written in Python, that is maintained by the Redbrick Webgroup. This project uses [`hikari`](https://github.com/hikari-py/hikari/), an opinionated microframework, to interface with the Discord API. [`hikari-arc`](https://github.com/hypergonial/hikari-arc) is the command handler of choice.
```
Comment (aydenjahola) [inline README.md:3]: I think adding hyperlink to what redbrick webgroup would be useful here

**Review by aydenjahola — CHANGES_REQUESTED** [review]
I think adding hyperlink to what redbrick webgorup is would be good if someone's viewing repo without knowing what webgroup is

**Review by wizzdom — DISMISSED** [review]
Looking good! Cleans things up nicely

**Review by HypnoAnt — DISMISSED** [review]
LGTM

**Review by aydenjahola — DISMISSED** [review]

**Review by aydenjahola — APPROVED** [review]

**Pull request merged**